### PR TITLE
implement product payments

### PR DESCRIPTION
**Pull request by ExtraGravity**

### CODE REVIEW

File: dbscripts/db-init.sql [hunk 1]
```diff
@@ -202,6 +202,7 @@ CREATE TABLE LoanPayments (
 -- ProductPayments
 
 CREATE TABLE ProductPayments (
+	productPaymentUuid UUID PRIMARY KEY NOT NULL DEFAULT uuid_generate_v4(),
```
Comment (beaesguerra) [inline dbscripts/db-init.sql:205]: what's the difference between this and productTransactionUuid?

File: src/database/ProductPayments/index.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      tableNames.MONEY_TRANSACTIONS + '.frompersonuuid',
+      'fromperson.personuuid')
+     .join(tableNames.PEOPLE + ' as toperson',
+      tableNames.MONEY_TRANSACTIONS + '.topersonuuid',
+      'toperson.personuuid')
+    .orderBy('fromlastname', 'asc')
+    .orderBy('datetime', 'asc');
+  },
+
+  /** inserts a MoneyuctTransaction row in the database */
```
Comment (beaesguerra) [inline src/database/ProductPayments/index.ts]: moneyuct transaction hahah

File: src/routers/transactions/money/productPayments.ts [hunk 3]
```diff
@@ -1,24 +1,177 @@
...
 router.get('/', async (req, res) => {
-  // TODO
-  res.status(StatusCode.OK).send('Successfully retrieved all product payments') ;
-}, authorized(UserType.ADMIN));
+  const result = await ProductPaymentsService.getProductPaymentsFromDb();
+  res.status(StatusCode.OK).send(result);
+}, authorized(UserType.ADMIN, UserType.TRADER, UserType.MONITOR));
 
+/**
+ * @api {post} /laons
```
Comment (beaesguerra) [inline src/routers/transactions/money/productPayments.ts]: typo

File: dbscripts/db-init.sql (hunk 1 above)
Comment (ExtraGravity) [reply dbscripts/db-init.sql:205]: this is a unique identifier on this table. productTransactionUuid relates it to one product transaction.

File: src/database/connection.ts [hunk 4]
```diff
@@ -47,6 +47,7 @@ export const tableNames = {
   MONEY_TRANSACTIONS: 'moneytransactions',
   LOANS: 'loans',
   PRODUCT_PAYMENTS: 'productpayments',
+  MEMOS: 'memos',
```
Comment (beaesguerra) [inline src/database/connection.ts]: is this memos change supposed to be here?

Comment (beaesguerra) [inline src/database/connection.ts]: is this memos change supposed to be here?

**Review by beaesguerra — APPROVED** [review]

File: src/database/connection.ts (hunk 4 above)
Comment (ExtraGravity) [reply src/database/connection.ts]: I think i branched off the wrong branch :P

**Pull request merged**